src/main/java/graphql/servlet/GraphQLServlet.java

```diff
@@ -260,17 +263,33 @@ private void doRequest(HttpServletRequest request, HttpServletResponse response,
             runCallbacks(requestCallbacks, c -> c.onError(request, response, t));
         } finally {
             runCallbacks(requestCallbacks, c -> c.onFinally(request, response));
+            if(asyncContext !=null)
+            	asyncContext.complete();
         }
     }
 
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        doRequest(req, resp, getHandler);
-    }
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		if (asyncServletMode) {
+			AsyncContext asyncContext = req.startAsync();
+			HttpServletRequest request = (HttpServletRequest) asyncContext.getRequest();
+			HttpServletResponse response = (HttpServletResponse) asyncContext.getResponse();
+			new Thread(() -> doRequest(request, response, getHandler, asyncContext)).start();
+		} else {
+			doRequest(req, resp, getHandler, null);
+		}
+	}
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        doRequest(req, resp, postHandler);
+    	if (asyncServletMode) {
+			AsyncContext asyncContext = req.startAsync();
+			HttpServletRequest request = (HttpServletRequest) asyncContext.getRequest();
+			HttpServletResponse response = (HttpServletResponse) asyncContext.getResponse();
+			new Thread(() -> doRequest(request, response, getHandler, asyncContext)).start();
```

Comment:
Should this be `postHandler` not `getHandler`.

Reply:
it seems you were right, it was a typical copy paste error. But its fixed in the merged version.